### PR TITLE
Make fonts available for older versions

### PR DESCRIPTION
**Pull request by Docteh**
All of the changes to the fonts have been backwards compatible, except for the recent move of crosshair

I've just taken the latest font files, and copied
0x7B to 0x26
0x7D to 0x27

### CODE REVIEW

**Comment by Docteh** [comment]
These modified fonts are explicitly for the already released versions of betaflight, like 4.0.x and earlier.  I've taken the nice clean fonts, and messed them up as little as possible. 

https://github.com/betaflight/betaflight/blob/4.0.x-maintenance/src/main/drivers/max7456_symbols.h#L56-L58
```
// AH Center screen Graphics
#define SYM_AH_CENTER_LINE          0x26
#define SYM_AH_CENTER_LINE_RIGHT    0x27
```



I didn't stick the current fonts into `resources/osd/v2` because I was imagining that the current version would be in `resources/osd` and anything in like `resources/osd/v1` or `resources/osd/v2` would be old stuff to ignore.

**Comment by mikeller** [comment]
Now I see. I guess there is no harm in this. But I think in this case we should also display the font version (which should ideally be numeric) in the font preview dialog, otherwise users will get confused about seeing different fonts when they open it for different firmware versions.
So how about we stick the current version fonts into `/resources/osd/2`, and the legacy version into `/resources/osd/1`?

**Comment by Docteh** [comment]
I guess the preview could say like `Version: 1` or `Version: 2` if selected from the dropdown (also when user loads the screen), and then just have it say `Version: custom` if the user loads a font from the dialog box?

**Comment by Docteh** [comment]
I'd like to limit the time I spend being mad at HTML, so I just to make sure you mean this preview: 


image

Busy,  but the version number is closer to the element where it matters...
image
Also I wouldn't have to worry about the scroll bar. 


I've updated the PR to move the current fonts into `resources/osd/2` and these legacy ones are `resources/osd/1`

- [x] Fonts into locations as discussed
- [ ] Figure out where exactly to put the version label

**Comment by mikeller** [comment]
I like the first version (version info above the font).

I would maybe extend it a bit like 'Font version: 1 (firmware version <= 4.0)'.

> time I spend being mad at HTML

I know this from somewhere...

**Comment by Docteh** [comment]
Current status:

Open Font File sets this:
image

Otherwise
image


image

**Comment by mikeller** [comment]
Looking good!

**Comment by Docteh** [comment]
I think this change is good to go, github says `Requested changes must be addressed to merge this pull request.` But I'm not sure how to make it happy there.

**Comment by mikeller** [comment]
That's just GitHub waiting for me to review and approve - will do that when I get time.

File: src/js/tabs/osd.js [hunk 1]
```diff
@@ -2401,30 +2403,38 @@ TABS.osd.initialize = function (callback) {
...
+        SYM.AH_CENTER_LINE_RIGHT = 0x27;
+        if(semver.gte(CONFIG.apiVersion,"1.42.0")) {
+            SYM.AH_CENTER_LINE = 0x7B;
+            SYM.AH_CENTER_LINE_RIGHT = 0x7D;
+        }
+
         fontPresetsElement.change(function (e) {
             var $font = $('.fontpresets option:selected');
-            $.get('./resources/osd/' + $font.data('font-file') + '.mcm', function (data) {
+            var fontver = '1';
```
Comment (mikeller) [inline src/js/tabs/osd.js]: I'd use numbers here. They are coerced when concatenated with strings.

File: src/js/tabs/osd.js [hunk 2]
```diff
@@ -2401,30 +2403,38 @@ TABS.osd.initialize = function (callback) {
...
+            if (semver.gte(CONFIG.apiVersion,"1.42.0")) {
```
Comment (mikeller) [inline src/js/tabs/osd.js]: Please use space after punctuation, just like in natural languages.

File: src/js/tabs/osd.js [hunk 3]
```diff
@@ -2401,30 +2403,38 @@ TABS.osd.initialize = function (callback) {
...
+                fontver = '2';
+            }
+            $('.font-manager-version-info').text(i18n.getMessage('osdDescribeFontVersion'+fontver));
```
Comment (mikeller) [inline src/js/tabs/osd.js]: Please use space around operators (except for on the inside of braces).

Essentially, most of the rules in https://github.com/betaflight/betaflight/blob/master/docs/development/CodingStyle.md are applicable to JavaScript as well.

File: src/js/tabs/osd.js [hunk 4]
```diff
@@ -2401,30 +2403,38 @@ TABS.osd.initialize = function (callback) {
...
         FONT.initData();
 
+        // Some of these definitions are determined by version.
+        SYM.AH_CENTER_LINE = 0x26;
+        SYM.AH_CENTER_LINE_RIGHT = 0x27;
+        if(semver.gte(CONFIG.apiVersion, "1.42.0")) {
+            SYM.AH_CENTER_LINE = 0x7B;
+            SYM.AH_CENTER_LINE_RIGHT = 0x7D;
+        }
+
```
Comment (Docteh) [inline src/js/tabs/osd.js]: I'm not sure if  I should put a comment near the top of this where all the rest of SYM is defined, mentioning uh, wherever in the code this is. like `// some of these are changed in the initialization function below`

or maybe move all of the definitions down here

**Comment by Docteh** [comment]
Thanks for the code review, remembering to follow coding standards doesn't happen easily...

File: src/js/tabs/osd.js (hunk 4 above)
Comment (mikeller) [reply src/js/tabs/osd.js]: Good point, I think so.

Comment (Docteh) [reply src/js/tabs/osd.js]: Going with the comment at the top. Otherwise I'd have to give some consideration to the `var STICK_OVERLAY_SPRITE` array

**Pull request merged**